### DISCUSSION TITLE
RSS feeds?

### DISCUSSION THREAD

**Discussion by markcellus** [bluesky-social/atproto#1662] [How to do it?] [answered]
Sorry if I'm posting this in the wrong category. But where do I find the RSS feeds for Bluesky profiles? Thanks for any information.

**Comment by snarfed** [comment]
Feel free to try https://bluesky-atom.appspot.com/

**Reply by markcellus** [reply]
Thanks but looking for a solution that isnt tied to a login and doesnt require a password

**Comment by pfrazee** [comment]
This isnt supported currently

**Comment by bnewbold** [comment]
I think this would be a great feature to support, but as Paul mentions isn't currently supported. Maybe we can add it around the same time as making the web interface public.

**Reply by markcellus** [reply]
Oh great! Is there a roadmap I can follow to be notified of when the interface will be made public?

**Reply by bnewbold** [reply]
Like an RSS feed? :grimacing: 

Nope, not yet... but this specific features should provide such an interface!

**Reply by markcellus** [reply]
Oh no, I meant just a public API for bluesky. Then maybe can get openrss.org to possibly make RSS feeds from it.

**Reply by bnewbold** [reply]
The bsky "appview" run by Bluesky (the company) is public at api.bsky.app. Note that most developers with an account connect to the PDS instance, using auth, at `bsky.social`, but the appview (aggregator) serves a subset of endpoints with no authentication.

You'd probably want an "author feed" like:
https://api.bsky.app/xrpc/app.bsky.feed.getAuthorFeed?actor=bsky.app&limit=30&filter=posts_no_replies

Change the `actor=bsky.app` to another account (like `actor=bnewbold.net`), or use a stable long-term DID instead of a handle: `actor=did:plc:z72i7hdynmk6r22z27h6tvur` (accounts can change handles at any time)

**Reply by snarfed** [reply]
@markcellus this ^ inspired me to tweak https://granary.io/ to use the prod AppView instead of PDS if no app password is provided. It might now do what you want: convert individual Bluesky users' feeds to RSS or Atom, without auth. Enter a user's handle, choose `@self` instead of `@all`, and choose format `rss` or `atom`.

**Reply by singpolyma** [reply]
Hmm when I try this with granary now it says auth required

**Comment by snarfed** [comment, marked as answer]
These shipped today with https://blueskyweb.xyz/blog/12-21-2023-butterfly ! eg https://bsky.app/profile/snarfed.org 's is https://bsky.app/profile/did:plc:fdme4gb7mu7zrie7peay7tst/rss

**Reply by devinivy** [reply]
Your profile page also contains a `<link>` to the RSS feed, so in many RSS readers you can provide it the link directly to your profile page and it will properly find the feed :). One convenience there is that you don't necessarily need to look-up a profile's did to be able to access their RSS feed.

**Reply by Diti** [reply]
Any chance we could get Atom feeds as well? The way they handle dates (ISO 8601-like) and require language attributes is helpful.

**Reply by apiontek** [reply]
Is this only enabled for some profiles? It doesn't seem enabled for mine, or a few others I just checked, but the one linked above for snarfed works. Is there a setting an account needs to turn on to enable this?

**Reply by snarfed** [reply]
@apiontek probably the "hide from logged out users" setting.

**Comment by EldonSamuelson** [comment]
I would be interested in a way to produce rss feeds for Bluesky Feeds

i.e. https://bsky.app/profile/did:plc:dhoindtmxo5mjmbdlovmt6o4/rss does work, but https://bsky.app/profile/did:plc:dhoindtmxo5mjmbdlovmt6o4/feed/aaacvhxrmf2e6/rss (a feed following 3 hashtags/keywords) does not

**Comment by paulgmiller** [comment]
Has anyone looked at the opposite importing rss feeds atproto? Hmmm maybe this https://sperea.es/blog/bot-bluesky-rss

**Reply by snarfed** [reply]
[Bridgy Fed](https://fed.brid.gy/) does this too!
